### DISCUSSION TITLE
Problems pairing with xbox controller

### DISCUSSION THREAD

**Discussion by fmorton** [pybricks/support#1979] [Q&A]
I have been unable to pair an xbox controller. It is an off-brand controller, but looks like it came of the same assembly line as the real ones, so let me know if this is a problem.

Here is a video showing the steps I am taking. Thanks for any help.

https://www.youtube.com/watch?v=x1mqcsjViVQ

**Comment by BertLindeman** [comment]
Fannk,

I only have experience with linux and windows.
Sometimes if I had xbox controller connect problem, I would remove the bluetooth entry for it from my pc
and start the xbox controller before I start the pybricks program. Sometime it helped.

Could you also show the versions of your firmware and the browser?

Bert

**Comment by fmorton** [comment]
Hub is 1.4.0. 

Chrome is  131.0.6778.140 (Official Build) (x86_64)

**Reply by BertLindeman** [reply]
> Hub is 1.4.0.

This confuses me. 

if I run 
```python

from pybricks import version

print(f'version {version}')
```
I see on my beta primehub:
```('primehub', '3.6.0b2', 'ci-build-3647-v3.6.0b2-37-g7f1b7f84 on 2024-12-16')```

The standard firmware on code.pybricks.com is 3.5.0.

I probably interpret your version wrong :smile:
Bert

**Reply by fmorton** [reply]
Sorry, I misunderstood:

version ('primehub', '3.6.0b2', 'ci-release-81-v3.6.0b2 on 2024-10-15')

**Reply by BertLindeman** [reply]
Thank you. 

That is nice and recent.
In that case I cannot help you further and know the experts will take you further.

**Comment by laurensvalk** [comment]
Hi @fmorton - We have only tested the official Xbox Controllers. It's possible that these off-brand controllers send alternate messages that the hub currently doesn't understand. I think we check for the Microsoft vendor identifier, for example.

If we can relax the filter strictness to make it work with other controllers (assuming we don't have to make other modifications since we couldn't test them), that could be worth it.

If you have an Android phone, you could install the NRF Connect app to see what advertising data your controller broadcasts when it is in pairing mode. Could you share a screenshot of your findings? An iOS phone might work too, though I think you can see fewer details.

**Comment by fmorton** [comment]
Thanks, Laurens. I was just testing one controller in prep for a robotics class in January. Sounds like it would be smarter to use the microsoft brand. I already got a couple of them and they worked immediately, so we will just do that.

One surprise. Is the expected behavior for the controller to turn off after the program stops?

Thanks for your project. We have a bunch of kids using pybricks.

**Reply by laurensvalk** [reply]
Thanks for sharing - and thank you for being a supporter of the project!

Yes, the controller disconnects when the program ends. And that causes the Xbox Controller to turn off.
